app/javascript/src/components/ClientInvoices/Header.tsx

```diff
@@ -0,0 +1,172 @@
+import { useUserContext } from "context/UserContext";
+
+import SearchDropdown from "./InvoiceSearch/SearchDropdown";
+
+const Header = ({ params, setParams }) => {
+  const { isDesktop, handleOverlayVisibility } = useUserContext();
+  const [searchQuery, setSearchQuery] = useState<string>(params.query || "");
+  const [searchResult, setSearchResult] = useState<any[]>([]);
+  const [status, setStatus] = useState<InvoiceStatus>(InvoiceStatus.IDLE);
+  const [expandSearchBox, setExpandSearchBox] = useState<boolean>(false);
+  const [isShrinkingSearchBox, setIsShrinkingSearchBox] =
+    useState<boolean>(false);
+  const debouncedSearchQuery = useDebounce(searchQuery, 500);
+
+  useEffect(() => {
+    if (searchQuery) {
+      fetchSearchinvoices(debouncedSearchQuery);
+    } else {
+      setSearchResult([]);
+      setStatus(InvoiceStatus.IDLE);
+    }
+  }, [debouncedSearchQuery]);
+
+  useEffect(() => {
+    const pageContainerEl = document.getElementById("invoice-list-page");
+
+    if (pageContainerEl && !isDesktop) {
+      if (expandSearchBox || searchQuery) {
+        pageContainerEl.classList.add("h-90v", "overflow-hidden");
+        handleOverlayVisibility(true);
+      } else {
+        pageContainerEl.classList.remove("h-90v", "overflow-hidden");
+        handleOverlayVisibility(false);
+      }
+    }
+
+    return () => {
+      if (pageContainerEl) {
+        pageContainerEl.classList.remove("h-90v", "overflow-hidden");
+        handleOverlayVisibility(false);
+      }
+    };
+  }, [expandSearchBox, searchQuery, isDesktop]);
+
+  const fetchSearchinvoices = async debouncedSearchQuery => {
+    const SEARCH_SIZE = 5;
+    const searchParams = `invoices_per_page=${SEARCH_SIZE}&page=1&query=${debouncedSearchQuery}`;
+    setStatus(InvoiceStatus.LOADING);
+    try {
+      const {
+        data: { invoices },
+      } = await clientApi.invoices(searchParams);
+      setSearchResult(invoices);
+      setStatus(InvoiceStatus.SUCCESS);
+    } catch {
+      setStatus(InvoiceStatus.ERROR);
+    }
+  };
+
+  const onSearchClear = () => {
+    setSearchQuery("");
+    window.localStorage.setItem(LocalStorageKeys.INVOICE_SEARCH_PARAM, "");
+    setParams({ ...params, query: "" });
+  };
+
+  const onKeydownHandler = e => {
+    if (e.key === "Enter") {
+      window.localStorage.setItem(
+        LocalStorageKeys.INVOICE_SEARCH_PARAM,
+        searchQuery
+      );
+      setParams({ ...params, query: searchQuery });
+    }
+  };
+
+  const onSearchBlur = () => {
+    setExpandSearchBox(false);
+    setIsShrinkingSearchBox(true);
+    const timer = setTimeout(() => {
+      setIsShrinkingSearchBox(false);
+      clearTimeout(timer);
+    }, 300);
+  };
+
+  return (
+    <div className="relative mt-6 mb-3 flex flex-wrap items-center justify-between md:justify-start lg:justify-between">
+      {isDesktop && <h2 className="header__title">Invoices</h2>}
+      {isDesktop ? (
```

Comment:
```suggestion
      {isDesktop ? (
       <h2 className="header__title">Invoices</h2>
```

Move the header inside the ternary condition true block.